server.go

```diff
@@ -1976,9 +1972,28 @@ out:
 
 				// Potentially notify any getblocktemplate long poll clients
 				// about stale block templates due to the new transaction.
-				s.rpcServer.gbtWorkState.NotifyMempoolTx(s.txMemPool.LastUpdated())
+				s.rpcServer.gbtWorkState.NotifyMempoolTx(
+					s.txMemPool.LastUpdated())
 			}
 
+		case <-s.quit:
+			break out
+		}
+	}
+	s.wg.Done()
```

Comment:
I believe there will need to be  cleanup section here that drains the `s.relayNtfnChan` channel.  Otherwise shutdown could hang because queued items in the sender would block forever.

The following should do the trick:

```Go
	// Drain channels before exiting so nothing is left waiting around
	// to send.
cleanup:
	for {
		select {
		case <-s.relayNtfnChan:
		default:
			break cleanup
		}
	}
	s.wg.Done()
```